core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -4031,6 +4033,41 @@ private static RelNode groupIdRel(RelBuilder builder, boolean extra) {
     assertThat(root, hasTree(expected));
   }
 
+  /**
+   * Tests {@link RelBuilder#in} would throw exception if types of the ranges are not compatible.
+   */
+  @Test void testFilterInWithNotCompatibleArguments() {
+    try {
+      final RelBuilder b = RelBuilder.create(config().build());
+      final RelNode root = b.scan("EMP")
+          .filter(b.in(b.field(2), b.literal(1), b.literal(true)))
+          .build();
+      fail("expected error, got " + root);
+    } catch (ClassCastException e) {
+      // good
```

Comment:
Can we assert the exception message here ?

Reply:
Good question, I once assert the exception message here. But I found assert only success for mac, failed in Window/Linux because ClassCastException message is minor different when run in the different environment. 
So I only  check the exception  class  here.